### PR TITLE
Add Data.Type.Equality.Compat

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: base-compat-batteries/src/Data/Type/Equality/Compat.hs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  (:~~:)(..),
+#endif
+
+  -- * Working with equality
+  sym, trans, castWith, gcastWith,
+#if __GLASGOW_HASKELL__ >= 706
+  apply,
+#endif
+#if __GLASGOW_HASKELL__ >= 708
+  inner, outer,
```
Comment (RyanGlScott) [inline base-compat-batteries/src/Data/Type/Equality/Compat.hs]: As I note in https://github.com/phadej/some/pull/2/files/778469cc2f72b44862c2c12c93254e33ab22dd84..c296e33d0a76590f0e6fa6e18c4dadc32d1565ef#diff-533148f9e61bee9cf4d20acc82d92e23, we can actually provide implementations of `inner` and `outer` on more versions of GHC than this.

File: base-compat-batteries/base-compat-batteries.cabal [hunk 2]
```diff
@@ -118,6 +119,7 @@ library
       Data.STRef.Compat
       Data.String.Compat
       Data.Type.Coercion.Compat
+      Data.Type.Equality.Compat
```
Comment (RyanGlScott) [inline base-compat-batteries/base-compat-batteries.cabal]: You've added `Data.Type.Equality.Compat` to the `.cabal` file here, although there doesn't appear to be a corresponding file. Did you forget to add it?

**Comment by RyanGlScott** [comment]
This PR also needs to update the changelogs for `base-compat` and `base-compat-batteries`, when you get a chance to do so.

File: base-compat-batteries/base-compat-batteries.cabal (hunk 2 above)
Comment (phadej) [reply base-compat-batteries/base-compat-batteries.cabal]: There is, you commented on it?

File: base-compat-batteries/src/Data/Type/Equality/Compat.hs (hunk 1 above)
Comment (phadej) [reply base-compat-batteries/src/Data/Type/Equality/Compat.hs]: Done.

**Comment by phadej** [comment]
As `type-equality` is not released, I hope travis to turn green any moment now!

File: base-compat-batteries/CHANGES.markdown [hunk 3]
```diff
@@ -31,6 +31,8 @@
    [`base-compat` changelog](https://github.com/haskell-compat/base-compat/blob/master/base-compat/CHANGES.markdown#changes-in-????-????????)
    for more details.
 
+ - Add `Data.Type.Equality.Compat` module
```
[the remark this answers is not in the document]
Comment (phadej) [reply base-compat-batteries/CHANGES.markdown]: I went "lazy" approach, copying `Data.Type.Coercion.Compat` and `s/Coercion/Equality/`.

File: base-compat-batteries/src/Data/Type/Equality/Compat.hs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+#if __GLASGOW_HASKELL__ >= 702
+{-# LANGUAGE Trustworthy #-}
+#endif
+#if __GLASGOW_HASKELL__ >= 708
+{-# LANGUAGE ExplicitNamespaces #-}
+#endif
+module Data.Type.Equality.Compat (
+  -- * The equality types
+  (:~:)(..),
+#if __GLASGOW_HASKELL__ >= 802
```
[the remark this answers is not in the document]
Comment (phadej) [reply base-compat-batteries/src/Data/Type/Equality/Compat.hs]: Fixed

File: base-compat-batteries/src/Data/Type/Equality/Compat.hs [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE CPP, NoImplicitPrelude, PackageImports #-}
+#if __GLASGOW_HASKELL__ >= 702
+{-# LANGUAGE Trustworthy #-}
```
[the remark this answers is not in the document]
Comment (phadej) [reply base-compat-batteries/src/Data/Type/Equality/Compat.hs]: Added

**Pull request merged**